server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -95,6 +101,32 @@ private NLPResult analyzeWithNLP(String text) throws IOException {
     return nlpResults;
   }
 
+  /**
+   * Takes the given map of sentiments and creates a common weighted 
+   * sentiment according to the number of likes each comment receives
+   * out of the sum of the likes of the comments
+   *
+   * @param unweightedNLPMap the unweighted map of <like, nlp> pairs
+   * @return A weighted NLPResult object
+   */
+  private NLPResult createWeightedSentiment(HashMap<Integer, NLPResult> unweightedNLPMap) {
+      double totalLikes = 0;
+
+      for (Map.Entry<Integer, NLPResult> likesAndNLPResult : unweightedNLPMap.entrySet()) {
+          totalLikes += (double) likesAndNLPResult.getKey();
+      }
+
+      double weightedMagnitude = 0;
+      double weightedScore = 0;
+
+      for (Map.Entry<Integer, NLPResult> likesAndNLPResult : unweightedNLPMap.entrySet()) {
+          weightedMagnitude += (likesAndNLPResult.getKey() / totalLikes) * likesAndNLPResult.getValue().magnitude;
```

Comment:
Store the total number of likes in a variable instead of calling `likesAndNLPResult.getKey()`. This will improve code clarity.